src/amd/compiler/aco_instruction_selection.cpp

```diff
@@ -109,6 +109,28 @@ Temp get_ssa_temp(struct isel_context *ctx, nir_ssa_def *def)
    return Temp{id, rc};
 }
 
+void emit_v_add32(isel_context *ctx, Temp dst, Operand a, Operand b)
```

Comment:
Just a question: do we have constants somewhere? because you'd have to handle the restrictions as well (there or here)
Might be easier to just have Temp parameters.

Reply:
Not currently, but it is done for non-zero/one instance rate divisors in my wip vertex shader changes.

The restrictions should be handled fine there.